### DISCUSSION TITLE
Ferromagnetic sphere in the homogeneous magnetic field

### DISCUSSION THREAD

**Discussion by dizcza** [magpylib/magpylib#560] [General]
I'd like to model the impact of a ferrite placed in the homogeneous Earth magnetic field. For this, I need to 
1. Place a ferrite of the given shape at the specified location.
2. Add homogeneous Earth magnetic field
3. Get the magnetic tensor B(x,y) and plot it like in the figure below or
4. Add a sensor that moves left to right and measures the magnetic field vector with time (position of the sensor).

I need to model a scene with several metallic objects in the Earth field to test a magnetometer. Is magpylib suitable for such modeling tasks?

![Screenshot from 2022-08-30 12-09-00](https://user-images.githubusercontent.com/7688337/187397610-fbc91178-66b7-4528-a5c3-0153f4998dae.png)

**Comment by OrtnerMichael** [comment]
Hi Danylo,

Soft-magnetic materials can in principle be modeled with Magpylib, but it requires a lot of know-how and hands-on work. You can use the so-called Magnetostatic Method of Moments for this purpose - you'll find a reference on Google Schoolar.

But why would you test a magnetometer using soft-magnetic bodies ? For this purpose you should use a Helmholtz Coil, or a very well-known permanent magnet, ideally in shielded environments.

sincerely,
Michael

**Reply by dizcza** [reply]
Sorry for the confusion, the better way to say it is that I'm looking for how metal objects change the magnetic field they are placed in to have an intuition of how accurate the (passive) magnetometer should be to capture the change of the vector **B** at 30 cm distance.

I see that it's not easily solved with magpylib.

Thanks.

**Reply by OrtnerMichael** [reply]
no its not trivial unfortunately.
